library/src/test/java/com/novoda/downloadmanager/StubCursor.java

```diff
@@ -0,0 +1,292 @@
+package com.novoda.downloadmanager;
+
+import android.content.ContentResolver;
+import android.database.CharArrayBuffer;
+import android.database.ContentObserver;
+import android.database.Cursor;
+import android.database.DataSetObserver;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class StubCursor implements Cursor {
```

Comment:
Plan to swap this out with a wrapped `Cursor` that we own and is easier to mock / rely on.

Reply:
Makes sense. There is a heck of a lot on Cursor.

Wish it didn't make sense. It's a shame Android deprecated MockCursor and just said "use Mockito", rather than providing a decent helpful fake.